src/phoneField/index.tsx

```diff
@@ -0,0 +1,179 @@
+import React, { PureComponent } from 'react'
+import cc from 'classcat'
+import prefix from '_utils'
+import isEmpty from 'lodash.isempty'
+import isEqual from 'lodash.isequal'
+import SelectField from 'selectField'
+import TextField from 'textField'
+import countryTelData from 'country-telephone-data'
+import style from 'phoneField/style'
+
+const FIELDS: phoneFields = {
+  PHONENUMBER: 'phoneNumber',
+  PHONEPREFIX: 'phonePrefix',
+}
+
+interface PhoneFieldOnChangeParameters {
+  name: string
+  value: object
+}
+interface PhoneFieldProps  {
+  readonly name: string
+  readonly id?: string
+  readonly className?: string
+  readonly defaultPrefix?: string
+  readonly prefixLabel?: string
+  readonly title?: string
+  readonly ariaLabelledBy?: string
+  readonly placeholder?: string
+  readonly defaultValue?: string
+  readonly countriesWhiteList?: string[]
+  readonly customCountriesName?: customCountriesName
+  readonly isInline?: boolean
+  readonly onChange?: (obj: PhoneFieldOnChangeParameters) => void
+}
+
+interface PhoneFieldState {
+  countriesData: object[]
+  countriesWhiteList: string[]
+  completePhoneNumber: string
+  // Make the type of [key: string]: string etc... assignable for setState()
+  [key: string]: any
+}
+
+const filterIso2 = (countriesList: string[]) => countryTelData.allCountries.filter(
```

Comment:
Maybe you should add some unit tests for this methods and the following ones.

Reply:
@nickoola

Reply:
Those functions are really specifics to the components.
And it's tested in unit tests on `countriesWhiteList`.